packages/ui/src/bounty/modals/WithdrawalStakeModal/WithdrawStakeModal.tsx

```diff
@@ -0,0 +1,102 @@
+import { useMachine } from '@xstate/react'
+import BN from 'bn.js'
+import React, { useEffect, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { useMyAccounts } from '@/accounts/hooks/useMyAccounts'
+import { useTransactionFee } from '@/accounts/hooks/useTransactionFee'
+import { InsufficientFundsModal } from '@/accounts/modals/InsufficientFundsModal'
+import { accountOrNamed } from '@/accounts/model/accountOrNamed'
+import { SuccessTransactionModal } from '@/bounty/modals/SuccessTransactionModal'
+import { withdrawalStakeMachine, WithdrawalStakeStates } from '@/bounty/modals/WithdrawalStakeModal/machine'
+import { WithdrawSignModal } from '@/bounty/modals/WithdrawSignModal'
+import { FailureModal } from '@/common/components/FailureModal'
+import { WaitModal } from '@/common/components/WaitModal'
+import { useApi } from '@/common/hooks/useApi'
+import { useModal } from '@/common/hooks/useModal'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+
+import { WithdrawStakeModalCall } from '.'
+
+export const WithdrawStakeModal = () => {
+  const { api, connectionState } = useApi()
+  const { t } = useTranslation('bounty')
+  const { hideModal, modalData } = useModal<WithdrawStakeModalCall>()
+
+  const [state, send] = useMachine(withdrawalStakeMachine)
+
+  const { active: activeMember } = useMyMemberships()
+  const { allAccounts } = useMyAccounts()
+
+  const transaction = useMemo(() => {
+    if (api && connectionState === 'connected' && activeMember) {
+      return api.tx.bounty.withdrawFunding({ Member: activeMember.id }, modalData.bounty.id)
+    }
+  }, [JSON.stringify(activeMember), connectionState])
+
+  const entry = useMemo(
+    () => modalData.bounty.entries?.find((entry) => entry.worker.id === activeMember?.id),
+    [activeMember?.id]
+  )
+
+  const feeInfo = useTransactionFee(activeMember?.controllerAccount, transaction)
+
+  useEffect(() => {
+    if (state.matches(WithdrawalStakeStates.requirementsVerification)) {
+      if (transaction && feeInfo && activeMember) {
+        feeInfo.canAfford && send('NEXT')
```

Comment:
`feeInfo.canAfford ? send('NEXT') : send('ERROR')` would be more straightforward, but it's only cosmetic